File: 0973-k-closest-points-to-origin/0973-k-closest-points-to-origin.py
import heapq

class Solution:
    def kClosest(self, points: List[List[int]], k: int) -> List[List[int]]:
        
        minHeap = []
        
        for i, j in points:
            distanceToOrigin = (i ** 2) + (j ** 2)
            minHeap.append([distanceToOrigin, i, j])
            
        heapq.heapify(minHeap)
        result = []
        
        while k > 0:
            distance, i, j = heapq.heappop(minHeap)
            result.append([i, j])
            k -= 1
            
        return result

File: 0973-k-closest-points-to-origin/0973-k-closest-points-to-origin.cpp
class Solution {
public:
    vector<vector<int>> kClosest(vector<vector<int>>& points, int k) {
        
        vector<vector<int>> minHeap;
        for(auto& p : points){
            int x = p[0], y = p[1];
            minHeap.push_back({-x*x - y*y, x, y});
        }
        make_heap(minHeap.begin(), minHeap.end());
        vector<vector<int>> ans;
        
        while(k--){
            vector<int> top = minHeap[0];
            pop_heap(minHeap.begin(), minHeap.end());
            minHeap.pop_back();
            
            ans.push_back({top[1], top[2]});
        }
        return ans;
    }
};